cpu/stm32_common/periph/flashpage.c

```diff
@@ -41,46 +62,65 @@ void flashpage_write(int page, void *data)
 
     /* unlock the flash module */
     DEBUG("[flashpage] unlocking the flash module\n");
-    if (FLASH->CR & FLASH_CR_LOCK) {
-        FLASH->KEYR = FLASH_KEY1;
-        FLASH->KEYR = FLASH_KEY2;
+    if (CNTRL_REG & CNTRL_REG_LOCK) {
+        KEY_REG = FLASH_KEY1;
+        KEY_REG = FLASH_KEY2;
     }
 
+#if defined(CPU_FAM_STM32L0)
+    DEBUG("[flashpage] unlocking the flash program memory\n");
+    if (!(CNTRL_REG & CNTRL_REG_LOCK)) {
+        if (FLASH->PECR & FLASH_PECR_PRGLOCK) {
```

Comment:
This nested if's seem a bit weird to me... especially because both `CTRL_REG` and `FLASH->PECR` are defining the same registers.